compiler+runtime/src/cpp/jank/runtime/ns.cpp

```diff
@@ -60,19 +72,29 @@ namespace jank::runtime
     return { expect_object<var>(*found) };
   }
 
-  result<void, native_persistent_string> ns::add_alias(obj::symbol_ptr const &sym, ns_ptr const &ns)
+  result<void, native_persistent_string>
+  ns::add_alias(obj::symbol_ptr const &sym, ns_ptr const &nsp)
   {
     auto locked_aliases(aliases.wlock());
     auto const found((*locked_aliases)->data.find(sym));
-    if(found && expect_object<var>(*found) != ns)
+    if(found)
     {
-      return err(fmt::format("Alias already bound to a different ns: {}", sym->to_string()));
+      if(expect_object<ns>(*found) != nsp)
```

Comment:
I'm confused about this. Why were we expecting a `var` first, now we're expecting an `ns`?

Reply:
`nsp` is an `ns_ptr` here so we need to cast to match it.

I improved the error message here, basing it on a similar message from `ns::refer`.

```clojure
Bottom of clojure.core
clojure.core=> (create-ns 'foo)
foo

clojure.core=> (alias 'a 'clojure.core)
nil

clojure.core=> (alias 'a 'clojure.core)
nil

clojure.core=> (alias 'a 'foo)
Exception: a already aliases clojure.core in ns clojure.core, cannot change to foo
```

Reply:
I don't think that addresses my confusion. We went from this: `expect_object<var>(*found)` to this: `expect_object<ns>(*found)`

Has the type of `*found` changed? Was it wrong to cast it to `var` in the first place?

Reply:
My best guess is you're looking at code that uses `this->vars`. This function uses `this->alias`.

Reply:
Oh, the diff. I can't explain that, I think it was wrong before.

Reply:
Switched to side-by-side view for this screenshot. Left side uses `expect_object<var>`. Right side uses `expect_object<ns>`.

![image](https://github.com/user-attachments/assets/d08e8abc-5824-46b0-90d1-d8b3aa6b9327)

Reply:
Yes, it crashes on main.

```clojure
Bottom of clojure.core
clojure.core=> (alias 'a 'clojure.core)
nil

clojure.core=> (alias 'a 'clojure.core)
Assertion failed: (o->type == T::obj_type), function expect_object, file rtti.hpp, line 91.
zsh: abort      ./build/jank repl
```

Reply:
> Switched to side-by-side view for this screenshot. Left side uses expect_object<var>. Right side uses expect_object<ns>.

Yes sorry. Left side contains a bug that I didn't notice.

Reply:
:joy: Alright. Agreed that it looks blatantly wrong on main.